### ISSUE TITLE
fix: Image does not include the newest patches for glibc

### ISSUE THREAD

**Issue by Nico-DB** [oracle/weblogic-image-tool#337]
When building a new image for (12.2.1.4.0) using --latestPSU AND --recommendedPatches there are still 3 finding within the image (scanned via Clair):
ELSA-2021-9344	glibc:2.28-164.0.1.el8	HIGH	
ELSA-2021-9344	glibc-common:2.28-164.0.1.el8	HIGH	
ELSA-2021-9344	glibc-minimal-langpack:2.28-164.0.1.el8	HIGH	

Could you please check if the base-image is updated correctly (yum update -y)

My complete call is the following:
bash imagetool.sh create --tag "${IMAGE_REPO_NAME}:${WEBLOGIC_VERSION}-domain" --jdkVersion $jdkFolder --version $WEBLOGIC_VERSION --latestPSU --recommendedPatches --user $oracleSupportUser --passwordFile $path/pw.txt --wdtVersion $deploytoolRelease --wdtDomainHome /u01/oracle/user_projects/domains/base_domain --wdtModel=$path/simple-topology.yaml --wdtVariables=$path/domain.properties --chown oracle:oinstall --additionalBuildCommands $path/additional_args --additionalBuildFiles

**Comment by jshum2479** [comment]
This is coming from the oel slim image, @mriccell

**Comment by ddsharpe** [comment]
@Nico-DB I will not have a chance to look into this until later this week.  glibc updates are not part of `--latestPSU` or `--recommendedPatches`.  Those two options are specific to WebLogic type patches.  glibc is updated through the OS command (yum) or by pulling the latest image. If you have the oraclelinux image downloaded already, the default Docker build behavior is to keep using the local copy.  The `--pull` command for Image Tool can be used to tell Docker to use its own pull option which will check the remote repository for an updated image of the same tag.

If the Oracle image for oraclelinux on the Oracle repository is not updated with the secure version of these libraries, you should file an issue on that image with Oracle.  If I find this to be the case later this week, I will file the issue there and post back here.

**Comment by ddsharpe** [comment]
@Nico-DB There doesn't appear to be anything ImageTool can do in this regard.  The released Oracle Linux image and the update repo are managed by the Oracle Linux team.  The released image and the repo only have glibc-2.28-164.0.1.el8.x86_64 as the latest.  The Image Tool is just consuming that image.  An additional update call results in the same version.  I sent a message over to the Oracle Linux team through Oracle Support, and suggest that you do the same so that you can be notified when they have an update.

**Comment by ddsharpe** [comment]
@Nico-DB The Oracle Linux team responded that they believe this to be a false positive.  The glibc bug was in an earlier version, but -164 does not have this bug.  They intend to update the doc.

**Comment by tvierling** [comment]
Oracle Linux ships two flavors of glibc and openssl packages, which are selectable by the user for different purposes:
* Standard or "vanilla" glibc/openssl
* glibc/openssl with enablement to be used with Ksplice Userspace (live-patching of userspace processes, marked by `ksplice` in the version string)

openssl also ships these two flavors, plus a third on OL7 specifically:
* openssl with FIPS validation (marked by `fips` in the version string)

Scanners that make use of Oracle Linux ELSA data, which are not using our OVAL definitions (which include conditionals to ensure only the selected flavor is checked for updates), must account for these differences, and only flag updates if they match the installed flavor.

https://linux.oracle.com/errata/ELSA-2021-9344.html is the Ksplice Userspace flavor of glibc on OL8. You can tell by the `ksplice` in the package version string.

Since containers don't support Ksplice Userspace, that means there's no point in trying to install that flavor of glibc; the scanner should be only checking against the "vanilla" flavor. The erratum for that flavor, for the same glibc version, is: https://linux.oracle.com/errata/ELSA-2021-1585.html

The bottom line here is that if you're using Clair and it's doing this, you may need to work with the Clair devs to implement proper recognition of these differing flavors of glibc/openssl RPMs.

btw @jshum2479 it's been called OL, not OEL, since 2010; please fix any docs that still refer to the ancient name ;)
